lib/ansible/modules/network/panos/panos_security_policy.py

```diff
@@ -0,0 +1,516 @@
+        default: true
+'''
+
+EXAMPLES = '''
+# permit ssh to 1.1.1.1
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'SSH permit'
+    description: 'SSH rule test'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['any']
+    application: ['ssh']
+    service: ['application-default']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# Allow HTTP multimedia only from CDNs
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'HTTP Multimedia'
+    description: 'Allow HTTP multimedia only to host at 1.1.1.1'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['content-delivery-networks']
+    application: ['http-video', 'http-audio']
+    service: ['service-http', 'service-https']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# more complex fictitious rule that uses profiles
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'Allow HTTP w profile'
+    log_start: false
+    log_end: true
+    action: 'allow'
+    antivirus: 'default'
+    vulnerability: 'default'
+    spyware: 'default'
+    url_filtering: 'default'
+    wildfire_analysis: 'default'
+    commit: false
+
+# deny all
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'DenyAll'
+    log_start: true
+    log_end: true
+    action: 'deny'
+    rule_type: 'interzone'
+    commit: false
+
+# permit ssh to 1.1.1.1 using panorama and pushing the configuration to firewalls
+# that are defined in 'DeviceGroupA' device group
+- name: permit ssh to 1.1.1.1 through Panorama
+  panos_security_policy:
+    ip_address: '10.5.172.92'
+    password: 'paloalto'
+    rule_name: 'SSH permit'
+    description: 'SSH rule test'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['any']
+    application: ['ssh']
+    service: ['application-default']
+    hip_profiles: ['any']
+    action: 'allow'
+    devicegroup: 'DeviceGroupA'
+'''
+
+RETURN = '''
+# Default return values
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import get_exception
+
+try:
+    import pan.xapi
+    from pan.xapi import PanXapiError
+    import pandevice
+    import pandevice.firewall
+    import pandevice.panorama
+    import pandevice.objects
+    import pandevice.policies
+
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
```

Comment:
Where are you using `HAS_LIB`?

Generally after `argspec` you'd do something like
```
if not HAS_LIB:
    module.fail_json(msg="Missing requried pan-python module (check module docs)")
```